Modules/posixmodule.c

```diff
@@ -13307,8 +13319,10 @@ ScandirIterator_iternext(ScandirIterator *iterator)
 
         /* Skip over . and .. */
         if (wcscmp(file_data->cFileName, L".") != 0 &&
-            wcscmp(file_data->cFileName, L"..") != 0) {
-            entry = DirEntry_from_find_data(&iterator->path, file_data);
+            wcscmp(file_data->cFileName, L"..") != 0)
+        {
+            PyObject *module = PyType_GetModule(Py_TYPE(iterator));
```

Comment:
This also only works because the iterator type is not subclassable.

Reply:
Hm, You are right. I use pep573 in functools. I got some error from test cases.

Something like:
======================================================================
ERROR: test_arg_combinations (test.test_functools.TestPartialCSubclass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/temp/shihai/cpython/Lib/test/test_functools.py", line 109, in test_arg_combinations
    p = self.partial(capture)
TypeError: PyType_GetModule: Type 'CPartialSubclass' has no associated module

https://github.com/python/cpython/pull/20055